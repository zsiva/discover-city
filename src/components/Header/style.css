.ui.pointing.secondary.menu {
  background-color: black;
}
.ui.pointing.secondary.menu .item {
  color: white;
}
