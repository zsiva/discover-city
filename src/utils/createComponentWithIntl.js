import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import messages from '../data/messages';
import flattenMessages from '../utils/flattenMessages';

const createComponentWithIntl = (children, props = { locale: 'en-US' }) => {
  return renderer.create(
    <IntlProvider {...props} messages={flattenMessages(messages[props.locale])}>
      {children}
    </IntlProvider>,
  );
};

export default createComponentWithIntl;
