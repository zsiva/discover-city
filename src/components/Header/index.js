//@flow
import React from 'react';
import { Icon, Menu } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import './style.css';

export default function Header() {
  return (
    <Menu inverted icon="labeled">
      <Link to="/city">
        <Menu.Item active={window.location.pathname === '/city'}>
          <Icon name="home" />
          City
        </Menu.Item>
      </Link>
      <Link to="/airport">
        <Menu.Item active={window.location.pathname === '/airport'}>
          <Icon name="plane" />
          Airport
        </Menu.Item>
      </Link>
      <Link to="/get-money">
        <Menu.Item active={window.location.pathname === '/get-money'}>
          <Icon name="money" />
          Casino
        </Menu.Item>
      </Link>
      <Link to="/user">
        <Menu.Item active={window.location.pathname === '/user'}>
          <Icon name="user" />
          Profile
        </Menu.Item>
      </Link>
    </Menu>
  );
}
