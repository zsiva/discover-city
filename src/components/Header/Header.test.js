import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import createComponentWithIntl from '../../utils/createComponentWithIntl';
import Header from './';

describe('Header', () => {
  it('should match snapshot, English', () => {
    const component = createComponentWithIntl(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
      { locale: 'en-US' },
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
