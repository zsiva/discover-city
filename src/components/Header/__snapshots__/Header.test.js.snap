// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot, English 1`] = `
<div
  className="ui inverted labeled icon menu"
>
  <div
    className="ui container"
  >
    <a
      href="/city"
      onClick={[Function]}
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="home small icon"
        />
        <span>
          City
        </span>
      </div>
    </a>
    <a
      href="/airport"
      onClick={[Function]}
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="plane icon"
        />
        <span>
          Airport
        </span>
      </div>
    </a>
    <a
      href="/police"
      onClick={[Function]}
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="building outline icon"
        />
        <span>
          Police
        </span>
      </div>
    </a>
    <a
      href="/get-money"
      onClick={[Function]}
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="money icon"
        />
        <span>
          Casino
        </span>
      </div>
    </a>
    <a
      href="/hotel"
      onClick={[Function]}
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="building icon"
        />
        <span>
          Hotel
        </span>
      </div>
    </a>
    <div
      className="right menu"
    >
      <a
        href="/user"
        onClick={[Function]}
      >
        <div
          className="item"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="user icon"
          />
          <span>
            Profile
          </span>
        </div>
      </a>
    </div>
  </div>
</div>
`;
